gt/main.c

```diff
@@ -1201,6 +1267,18 @@ fill_notify_pkt(struct ggu_policy *policy,
 			policy->params.declined.expire_sec);
 		break;
 	}
+	case GK_BPF: {
+		struct ggu_bpf_wire *bpf_wire_be = (struct ggu_bpf_wire *)
+			(ggu_decision->ip_flow + params_offset);
+		bpf_wire_be->expire_sec = rte_cpu_to_be_32(
+			policy->params.bpf.expire_sec);
+		bpf_wire_be->program_index = policy->params.bpf.program_index;
+		bpf_wire_be->reserved = 0;
+		bpf_wire_be->cookie_len_4by = cookie_len_4by;
+		rte_memcpy(bpf_wire_be->cookie, &policy->params.bpf.cookie,
```

Comment:
The parameters list in the BPF cookies part is possible in machine byte order instead of network order.

Reply:
It is intentional. Gatekeeper doesn't know how the cookie is used, so it pushes the responsibility of byte order to the BPF program itself. Currently, I'm leaving our BPF programs in little endian until we have demand for big endian as well. Anyways, if you ever need to support big endian in a BPF program, only the BPF program needs to support it, that is, Gatekeeper is not affected.

Reply:
I've added a comment to the code to explain it.